channeld/channeld.c

```diff
@@ -4046,6 +5884,11 @@ int main(int argc, char *argv[])
 			tptr = &timeout;
 		}
 
+		/* If we're in STFU mode and aren't waiting for a STFU mode
+		 * specific message, don't read from the peer-> */
+		if (!peer->stfu_wait_single_msg && is_stfu_active(peer))
+			FD_CLR(peer->pps->peer_fd, &rfds);
```

Comment:
This case should only fire when we're waiting on the user for something (ie. signed PSBT), and as such should only apply to the initiator side.

The acceptor side should be sitting idle waiting for us. If a message is sent during this time it should come through on the next time we expect a message (and at which point it will be the wrong type and we'll fail appropriately).

The only corner case I can think of is if we wanted to response to `tx_abort` earlier -- but waiting until the user finishes their task (ie. signing the PSBT) doesn't seem bad to me either. Hmmm 🤔

Reply:
Investigating this further: Yes I the splicing protocol will receive messages and notice the invalide one there.